### PR TITLE
Map and Set to have consistent XML serialization order

### LINKED ISSUE #1641: Feature request: Consistent serialization of XML to avoid unnecessary changes
Using OH 3.1.0.

Similar to https://github.com/openhab/openhab-core/issues/2436 (and related community discussion https://community.openhab.org/t/storing-configuration-userdata-in-git/124727/) which was discussing jsondb.

Currently the zwave bundle serializes ZWave node information in XML files. The way the serialization has been implemented, we are not guaranteed to get very stable ordering of entries. In my use case I see this clearly as I use git as a backup method for the `userdata` folder of openHAB -- as the XML ordering changes, the diffs are very hard to read, and real changes get lost in the noise. Specifically, entries of `<supportedCommandClasses class="concurrent-hash-map">` have a quite unpredictable serialization order.

We could serialize the hashmap in alphabetical order to avoid the issue.

I can contribute PR to this one but wanted to open up this issue first to get feedback on the suggestion.

### PR DESCRIPTION
**Pull request by ssalonen**
Resolves #1641

Signed-off-by: Sami Salonen <ssalonen@gmail.com>

### CODE REVIEW

File: src/main/java/org/openhab/binding/zwave/internal/OrderingSetConverter.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import com.thoughtworks.xstream.mapper.Mapper;
+
+/**
+ * Implements a converter which re-orders set on marshalling
+ *
+ * @author Sami Salonen
+ */
+public class OrderingSetConverter extends CollectionConverter {
+
+    public static boolean allSameClassAndComparable(Set<@Nullable ?> src) {
```
Comment (ssalonen) [inline src/main/java/org/openhab/binding/zwave/internal/OrderingSetConverter.java:34]: Copy-pasted from open openhab-core PR https://github.com/openhab/openhab-core/pull/2437. I think this is OK as both projects have same license and I am contributing the code in both cases.

https://github.com/openhab/openhab-core/blob/f68ca0c8d99b7938d6f7839b9f649cbd3919fb42/bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/OrderingSetSerializer.java#L35-L50

**Comment by ssalonen** [comment]
The tests asserting ordering fail on my machine when uncommenting additions in `ZWaveNodeSerializer`.

Short snippet from meld how the serialization changes

image

**Comment by ssalonen** [comment]
@cdjackson anything I can do to support in the review process?

**Comment by cdjackson** [comment]
Sorry - I did look over this some time back and through I'd merged already.

Thanks @ssalonen

**Pull request merged**